projects/core/components/hosted-dropdown/hosted-dropdown.component.ts

```diff
@@ -159,7 +159,7 @@ export class TuiHostedDropdownComponent implements TuiFocusableElementAccessor {
         }
     }
 
-    @HostListener('keydown.esc', ['$event'])
+    @HostListener('document:keydown.escape', ['$event'])
```

Comment:
This would cause change detection on every escape keypress on the document. This is a very common component, we definitely want to avoid that.

Reply:
fixed

Reply:
Maybe we can use `.silent` and `@shouldCall`? But they don't work with angular keydown effects, so we would need to filter for `escape` button inside `@shouldCall` predicate.

Reply:
`.silent` works as well

Reply:
It will not work, because Angular re-enters the zone in those keydown events, so it works, but it's not «silent».

Reply:
@waterplea you are right, sorry for that